### PR TITLE
🪲 [Fix]: Test data reaches module tests in every consumer repository

### LINKED ISSUE #378: Fix version resolution ignoring the bump label on pull requests
Modules built with the [Process-PSModule](https://github.com/PSModule/Process-PSModule) workflow signal
the intended release size by applying a version-bump label (`major`, `minor`, or `patch`) to a pull
request. While the pull request is open, the pipeline resolves and reports the version the change would
produce, so the author can confirm the next version before merging.

### What happens

On a pull request the resolved/previewed version always reflects a **patch** increment, regardless of
which bump label is applied. For a repository whose latest release is `v0.4.1`, a pull request labelled
`minor` previews `v0.4.2-<branch>` instead of `v0.5.0-<branch>`. The bump label appears to be ignored
until the pull request is merged into the default branch, so the author cannot see the version the change
will actually create.

### What is expected

The previewed version reflects the applied bump label: a `minor` pull request previews the next minor
version, a `major` pull request previews the next major version, and so on. Whether the run publishes
anything is a separate concern and should be controlled independently (by the `prerelease` label and the
merge event), not conflated with which version is shown.

### Reproduction steps

1. Use a module repository whose latest release is, for example, `v0.4.1`.
2. Open a pull request and apply the `minor` label (and no `prerelease` label).
3. Open the run's **Plan → Resolve-Version** step.
4. Observe the resolved version is `v0.4.2-<branch>NNN` (a patch), not `v0.5.0-<branch>NNN`.

### Environment

- **Process-PSModule:** v6.1.1 (also reproduces on the current default branch)
- **Event:** `pull_request`

### Regression

Not a recent regression. The bump label has only ever been evaluated on publishing events, so
non-publishing pull requests have always previewed a patch version.

### Workaround

Also apply a `prerelease` label: that switches the run to a prerelease-publishing event, which causes the
bump label to be evaluated. It has the side effect of actually publishing a prerelease, so it is not a
true preview.

### Acceptance criteria

- On a pull request, the previewed version reflects the applied `major`/`minor`/`patch` label.
- With no bump label, the preview still falls back to a patch version.
- The `prerelease` label controls only whether a prerelease is published — not which version is shown.
- Released version numbers on merge to the default branch are unchanged.

---

## Technical decisions

**Root cause:** The version-bump label evaluation lives in
[`PSModule/Resolve-PSModuleVersion`](https://github.com/PSModule/Resolve-PSModuleVersion), invoked by the
`Resolve-Version` step of
[`Plan.yml`](https://github.com/PSModule/Process-PSModule/blob/main/.github/workflows/Plan.yml). In
`Resolve-ReleaseDecision` the `major`/`minor`/`patch` evaluation was nested inside an
`if ($shouldPublish) { ... }` block. On a `pull_request` event,
[`Get-PSModuleSettings`](https://github.com/PSModule/Get-PSModuleSettings) sets
`Settings.Publish.Module.ReleaseType = 'None'` (a release is only cut on merge to the default branch, or a
prerelease when the `prerelease` label is present), so `ShouldPublish` is `$false`, the label block is
skipped, and the decision falls back to a patch bump.

**Fix approach:** Always evaluate the bump labels in `Resolve-ReleaseDecision`, independent of
`ShouldPublish`. `ReleaseType` (and the `prerelease` label that drives it) then controls only whether and
how the run publishes — never the version increment. The no-bump-label fallback continues to default to
patch and to not publish an unlabelled change.

**Fix location:** The change belongs in `Resolve-PSModuleVersion` (proposed in
[PSModule/Resolve-PSModuleVersion#9](https://github.com/PSModule/Resolve-PSModuleVersion/pull/9)).
Process-PSModule consumes it by bumping the pinned `Resolve-PSModuleVersion` reference in `Plan.yml` once
released.

**Backward compatibility:** All existing `Resolve-PSModuleVersion` decision, version and end-to-end test
cases are unchanged; the only behavioural difference is for label combinations that were previously never
evaluated (for example `minor` on a non-publishing event). Merge-to-default-branch releases are
unaffected.

**Verified:** With the fix wired in, a `minor`-labelled pull request against `v0.4.1` previews
`v0.5.0-<branch>001` with `ShouldPublish = False` (was `v0.4.2-<branch>001` before).

**Related:** #373 concerns a separate `Resolve-Version` gap on `schedule`/`workflow_dispatch` events. This
issue is specifically about bump labels being ignored on `pull_request` events.

---

### Resolve-PSModuleVersion

- [x] Move the `major`/`minor`/`patch` label evaluation out of the `if ($shouldPublish)` gate in
  `Resolve-ReleaseDecision` so it always runs ([#9](https://github.com/PSModule/Resolve-PSModuleVersion/pull/9))
- [x] Keep the no-bump-label fallback (default patch, do not publish an unlabelled change)
- [x] Add unit tests for `None + minor` and `None + major` previews
- [ ] Release a new `Resolve-PSModuleVersion` version

### Process-PSModule

- [ ] Bump the pinned `Resolve-PSModuleVersion` reference in `.github/workflows/Plan.yml` to the released version
- [ ] Confirm on a labelled pull request that the previewed version reflects the bump label

### PR DESCRIPTION
**Pull request by MariusStorhaug**
Module test workflows now receive the caller-provided test data (secrets and variables) in every repository that consumes Process-PSModule, and version resolution once again honors the bump label on pull requests. Previously the BeforeAll, Test, and AfterAll module jobs failed at the "Expose caller-provided test data" step because they ran a script that shipped only inside Process-PSModule's own repository, so no consumer could actually use the `TestData` secret introduced in v6.0.0.

- Fixes #378

## Fixed: Test data now reaches module tests in consumer repositories

The `TestData` secret (a single-line JSON object of `secrets` and `variables`) is now exposed to the BeforeAll, Test, and AfterAll module test jobs in any repository that consumes Process-PSModule. Each job installs the shared helpers and runs the `Import-TestData` command, which reads the payload, masks the secrets, and publishes every entry as an environment variable for the tests.

The consumer interface is unchanged — callers already pass `TestData` to `workflow.yml` (added in v6.0.0), so no migration is required:

```yaml
secrets:
  TestData: >-
    { "secrets": { "TEST_SECRET": "${{ secrets.TEST_SECRET }}" },
      "variables": { "TEST_VARIABLE": ${{ toJSON(vars.TEST_VARIABLE) }} } }
```

Secrets arrive masked in the logs; variables arrive verbatim.

## Fixed: Bump labels are honored during version resolution on pull requests

Version resolution now always evaluates the Major/Minor/Patch bump label, so a pull request labeled `Minor` previews the correct next minor version instead of defaulting to a patch prerelease. This is picked up by adopting `Resolve-PSModuleVersion` v1.1.4.

## Technical Details

- Removed `.github/scripts/Expose-TestData.ps1` (138 lines); the exposure logic moved into the shared `PSModule/Install-PSModuleHelpers` module as the `Import-TestData` command, so it no longer has to be checked out from the caller's repository.
- `BeforeAll-ModuleLocal.yml`, `Test-ModuleLocal.yml`, and `AfterAll-ModuleLocal.yml` each now install `PSModule/Install-PSModuleHelpers@v1.0.9` and run `Import-TestData` (env `PSMODULE_TEST_DATA: ${{ secrets.TestData }}`) in place of the deleted script.
- `Plan.yml`: `PSModule/Resolve-PSModuleVersion` pinned to v1.1.4, which contains the always-evaluate-labels fix.
- `Build-Site.yml`: `Install-PSModuleHelpers` bumped v1.0.8 → v1.0.9 so the action resolves to a single version across the repo.
- `README.md` documents that test data is exposed through `Install-PSModuleHelpers` / `Import-TestData`.
- All action references are pinned to release commit SHAs (IPH v1.0.9 = `8bfb84d557755c67d9b5643efe573bdcae4c1a4a`, Resolve v1.1.4 = `8d1dac7f326a45ba08060c1e24a5dd6f6f00b3ab`); actionlint is clean.
- Validated end-to-end on a consumer module: all module test matrix jobs (Linux/Windows/macOS) pass, with a masked secret and a verbatim variable observed in the tests.

### CODE REVIEW

File: .github/workflows/Test-ModuleLocal.yml [hunk 1]
```diff
@@ -36,21 +36,21 @@ jobs:
...
-
       - name: Download module artifact
         uses: actions/download-artifact@3e5f45b2cfb9172054b4087a40e8e0b5a5461e7c # v8.0.1
         with:
           name: module
           path: ${{ fromJson(inputs.Settings).WorkingDirectory }}/outputs/module
 
       - name: Install-PSModuleHelpers
-        uses: PSModule/Install-PSModuleHelpers@68e8ca76be679bfcb7099aed8cffa911c4ab72af # v1.0.8
+        uses: PSModule/Install-PSModuleHelpers@feat/import-testdata # E2E: temporary branch ref (Import-TestData not yet released); pin to a released SHA before merge
```
[the remark this answers is not in the document]
Comment (MariusStorhaug) [reply .github/workflows/Test-ModuleLocal.yml]: Addressed in d028861 — `PSModule/Install-PSModuleHelpers` is now pinned to the immutable release SHA `8bfb84d557755c67d9b5643efe573bdcae4c1a4a` (v1.0.9), replacing the temporary `@feat/import-testdata` branch ref. v1.0.9 is the release that ships the `Import-TestData` command this PR depends on, so pinning was intentionally deferred until it shipped. No mutable action refs remain in the workflows and CodeQL now passes.

File: .github/workflows/BeforeAll-ModuleLocal.yml [hunk 2]
```diff
@@ -31,12 +31,15 @@ jobs:
           persist-credentials: false
           fetch-depth: 0
 
+      - name: Install-PSModuleHelpers
+        uses: PSModule/Install-PSModuleHelpers@feat/import-testdata # E2E: temporary branch ref (Import-TestData not yet released); pin to a released SHA before merge
```
[the remark this answers is not in the document]
Comment (MariusStorhaug) [reply .github/workflows/BeforeAll-ModuleLocal.yml]: Addressed in d028861 — `PSModule/Install-PSModuleHelpers` is now pinned to the immutable release SHA `8bfb84d557755c67d9b5643efe573bdcae4c1a4a` (v1.0.9), replacing the temporary `@feat/import-testdata` branch ref. v1.0.9 is the release that ships the `Import-TestData` command this PR depends on, so pinning was intentionally deferred until it shipped. No mutable action refs remain in the workflows and CodeQL now passes.

File: .github/workflows/AfterAll-ModuleLocal.yml [hunk 3]
```diff
@@ -31,12 +31,15 @@ jobs:
           persist-credentials: false
           fetch-depth: 0
 
+      - name: Install-PSModuleHelpers
+        uses: PSModule/Install-PSModuleHelpers@feat/import-testdata # E2E: temporary branch ref (Import-TestData not yet released); pin to a released SHA before merge
```
[the remark this answers is not in the document]
Comment (MariusStorhaug) [reply .github/workflows/AfterAll-ModuleLocal.yml]: Addressed in d028861 — `PSModule/Install-PSModuleHelpers` is now pinned to the immutable release SHA `8bfb84d557755c67d9b5643efe573bdcae4c1a4a` (v1.0.9), replacing the temporary `@feat/import-testdata` branch ref. v1.0.9 is the release that ships the `Import-TestData` command this PR depends on, so pinning was intentionally deferred until it shipped. No mutable action refs remain in the workflows and CodeQL now passes.

File: .github/workflows/Test-ModuleLocal.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (MariusStorhaug) [reply .github/workflows/Test-ModuleLocal.yml]: Addressed in d028861 — `PSModule/Install-PSModuleHelpers` is now pinned to the immutable release SHA `8bfb84d557755c67d9b5643efe573bdcae4c1a4a` (v1.0.9), replacing the temporary `@feat/import-testdata` branch ref. v1.0.9 is the release that ships the `Import-TestData` command this PR depends on, so pinning was intentionally deferred until it shipped. No mutable action refs remain in the workflows and CodeQL now passes.

File: .github/workflows/BeforeAll-ModuleLocal.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (MariusStorhaug) [reply .github/workflows/BeforeAll-ModuleLocal.yml]: Addressed in d028861 — `PSModule/Install-PSModuleHelpers` is now pinned to the immutable release SHA `8bfb84d557755c67d9b5643efe573bdcae4c1a4a` (v1.0.9), replacing the temporary `@feat/import-testdata` branch ref. v1.0.9 is the release that ships the `Import-TestData` command this PR depends on, so pinning was intentionally deferred until it shipped. No mutable action refs remain in the workflows and CodeQL now passes.

File: .github/workflows/AfterAll-ModuleLocal.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (MariusStorhaug) [reply .github/workflows/AfterAll-ModuleLocal.yml]: Addressed in d028861 — `PSModule/Install-PSModuleHelpers` is now pinned to the immutable release SHA `8bfb84d557755c67d9b5643efe573bdcae4c1a4a` (v1.0.9), replacing the temporary `@feat/import-testdata` branch ref. v1.0.9 is the release that ships the `Import-TestData` command this PR depends on, so pinning was intentionally deferred until it shipped. No mutable action refs remain in the workflows and CodeQL now passes.

File: .github/workflows/Plan.yml [hunk 4]
```diff
@@ -85,7 +85,7 @@ jobs:
           ImportantFilePatterns: ${{ inputs.ImportantFilePatterns }}
 
       - name: Resolve-Version
-        uses: PSModule/Resolve-PSModuleVersion@d53326c7687d20a7949d457fccd71223856b1890 # v1.1.0
+        uses: PSModule/Resolve-PSModuleVersion@feat/always-evaluate-version-labels # E2E: temp branch ref (always-evaluate-labels); pin to a released SHA before merge
         id: Resolve-Version
         env:
```
[the remark this answers is not in the document]
Comment (MariusStorhaug) [reply .github/workflows/Plan.yml:90]: Addressed in d028861 — `PSModule/Resolve-PSModuleVersion` is now pinned to the immutable release SHA `8d1dac7f326a45ba08060c1e24a5dd6f6f00b3ab` (v1.1.4), replacing the `@feat/always-evaluate-version-labels` branch ref. v1.1.4 is the release containing the always-evaluate-labels fix, so this now matches the repo's SHA-pinning convention.

File: .github/workflows/Plan.yml [hunk 5]
```diff
@@ -85,7 +85,7 @@ jobs:
           ImportantFilePatterns: ${{ inputs.ImportantFilePatterns }}
 
       - name: Resolve-Version
-        uses: PSModule/Resolve-PSModuleVersion@d53326c7687d20a7949d457fccd71223856b1890 # v1.1.0
+        uses: PSModule/Resolve-PSModuleVersion@feat/always-evaluate-version-labels # E2E: temp branch ref (always-evaluate-labels); pin to a released SHA before merge
```
[the remark this answers is not in the document]
Comment (MariusStorhaug) [reply .github/workflows/Plan.yml]: Addressed in d028861 — the ref is no longer a temporary branch; it is pinned to the released `Resolve-PSModuleVersion` v1.1.4 SHA `8d1dac7f326a45ba08060c1e24a5dd6f6f00b3ab`. This change is in scope and intentional: it is the fix for #378 (bump labels ignored during version resolution on pull requests), and it is documented in the PR description under "Fixed: Bump labels are honored during version resolution on pull requests" and in Technical Details. Keeping it.

File: .github/workflows/Test-ModuleLocal.yml [hunk 6]
```diff
@@ -36,21 +36,21 @@ jobs:
...
-
       - name: Download module artifact
         uses: actions/download-artifact@3e5f45b2cfb9172054b4087a40e8e0b5a5461e7c # v8.0.1
         with:
           name: module
           path: ${{ fromJson(inputs.Settings).WorkingDirectory }}/outputs/module
 
       - name: Install-PSModuleHelpers
-        uses: PSModule/Install-PSModuleHelpers@68e8ca76be679bfcb7099aed8cffa911c4ab72af # v1.0.8
+        uses: PSModule/Install-PSModuleHelpers@8bfb84d557755c67d9b5643efe573bdcae4c1a4a # v1.0.9
```
[the remark this answers is not in the document]
Comment (MariusStorhaug) [reply .github/workflows/Test-ModuleLocal.yml:46]: Addressed — the PR description no longer contains a temporary-branch-ref note. It now states that all action references are pinned to release commit SHAs (`Install-PSModuleHelpers` v1.0.9 = `8bfb84d557755c67d9b5643efe573bdcae4c1a4a`, `Resolve-PSModuleVersion` v1.1.4 = `8d1dac7f326a45ba08060c1e24a5dd6f6f00b3ab`). Description and workflow now match — `Test-ModuleLocal.yml` line 46 pins v1.0.9 — so there is nothing left to reconcile.
